### PR TITLE
*-boilerplate: set a deadline for listener.Accept

### PR DESCRIPTION
**Pull request by mdibaiee**
**Description:**

- Set a timeout for our listening on connector side for a connection to be initiated, otherwise connector processes can be stuck even when sent a SIGTERM, as was observed by @dhammika 

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/391)

### CODE REVIEW

File: source-boilerplate/boilerplate.go [hunk 1]
```diff
@@ -21,17 +22,27 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
+const LISTEN_TIMEOUT_SECONDS = 30
```
Comment (willdonnelly) [inline source-boilerplate/boilerplate.go:25]: Is there a reason the timeout is different for sources vs materializations?

File: materialize-boilerplate/boilerplate.go [hunk 2]
```diff
@@ -25,17 +26,25 @@ type Options struct {
 	LogLevel string `long:"log.level" description:"Log level, one of: error, warn, info, debug, trace" default:"info"`
 }
 
+const LISTEN_TIMEOUT_SECONDS = 10
```
Comment (willdonnelly) [inline materialize-boilerplate/boilerplate.go:29]: nit: Idiomatic capitalization for constants in Go is is CamelCase.

File: source-boilerplate/boilerplate.go [hunk 3]
```diff
@@ -21,17 +22,27 @@ import (
...
 // RunMain is the boilerplate main function of a capture connector.
 func RunMain(srv pc.DriverServer) {
 	var logConfig = &logConfig{}
 	var parser = flags.NewParser(logConfig, flags.Default)
 	var ctx, _ = signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT)
 
 	var conn net.Conn
-	if server, err := net.Listen("tcp", ":2222"); err != nil {
+	server, err := net.ListenTCP("tcp4", &net.TCPAddr{
+		IP:   net.IPv4(127, 0, 0, 1),
```
Comment (willdonnelly) [inline source-boilerplate/boilerplate.go:35]: Likewise, is there a reason sources listen on `127.0.0.1` vs materializations on `0.0.0.0`? It seems like the external-reachability constraints should be the same for both.

**Review by willdonnelly — APPROVED** [review]
LGTM in spirit, but wondering about the differences between source and materialization versions.

**Review by williamhbaker — APPROVED** [review]
Similar comments as @willdonnelly regarding the differences between source and materialization timeout and address, other than that LGTM.

File: source-boilerplate/boilerplate.go (hunk 1 above)
Comment (mdibaiee) [reply source-boilerplate/boilerplate.go]: Sorry I was going back and forth between testing this PR and another investigation, they are meant to be the same, I just reconsolidated them

**Pull request merged**